sys/kern/syscalls.master

```diff
@@ -388,15 +416,15 @@
 	}
 53	AUE_SIGALTSTACK	STD|CAPENABLED {
 		int sigaltstack(
-		    _In_opt_ const stack_t *ss,
-		    _Out_opt_ stack_t *oss
+		    _In_opt_ _Contains_long_ptr_ const struct sigaltstack *ss,
+		    _Out_opt_ _Contains_long_ptr_ struct sigaltstack *oss
 		);
 	}
 54	AUE_IOCTL	STD|CAPENABLED {
 		int ioctl(
 		    int fd,
 		    u_long com,
-		    _Inout_opt_ char *data
+		    _Inout_opt_ _Contains_long_ptr_timet_ char *data
```

Comment:
Same question, why Contains is needed for data?

Reply:
So the fact that ioctl requires handling is detected by makesyscalls.lua. `data` can and does point to anything.

Reply:
To elaborate, I'd prefer not to require additions to syscalls.conf to indicate that syscalls need ABI-specific handling. Some can't seem to be avoided for freebsd32 (they all happen to be masked on CheriBSD because all syscalls for integer-pointer ABIs require handling.)

Reply:
I've decided to drop `time_t_`. It seems likely such an ioctl exists, but freebsd32_ioctl only handles ioctl commands that contain longs and pointers and those likely should be pushed deeper into the kernel where there isn't a risk (however small) of the command being conflated with another one.